scarlet/observation.py

```diff
@@ -149,6 +138,22 @@ def match(self, model_frame, diff_kernels=None, convolution="fft"):
             if type(self.weights) is np.ndarray:
                 self.weights = self.weights.copy().astype(model_frame.dtype)
 
+        # determine which channels are covered by data
+        # this has to be done first because convolution is only determined for these
+        self._channel_map = self.get_channel_map_for(model_frame)
```

Comment:
We could do that for the channel information in the future. But I wouldn't do it for entire `Frame` instances because we need matching in spatial and spectral coordinates, and I expect that we can't do that in one go.